crates/ty_server/src/server.rs

```diff
@@ -47,23 +49,40 @@ impl Server {
         in_test: bool,
     ) -> crate::Result<Self> {
         let (id, init_value) = connection.initialize_start()?;
-        let init_params: InitializeParams = serde_json::from_value(init_value)?;
-
-        let AllOptions {
-            global: global_options,
-            workspace: mut workspace_options,
-        } = AllOptions::from_value(
-            init_params
-                .initialization_options
-                .unwrap_or_else(|| serde_json::Value::Object(serde_json::Map::default())),
-        );
 
-        let client_capabilities = init_params.capabilities;
+        let InitializeParams {
+            initialization_options,
+            capabilities: client_capabilities,
+            workspace_folders,
+            ..
+        } = serde_json::from_value(init_value)?;
+
+        let (initialization_options, deserialization_error) =
+            InitializationOptions::from_value(initialization_options);
+
+        if !in_test {
+            crate::logging::init_logging(
+                initialization_options.log_level.unwrap_or_default(),
+                initialization_options.log_file.as_deref(),
+            );
+        }
+
+        tracing::info!("Initialization options: {initialization_options:?}");
+
+        if let Some(error) = deserialization_error {
+            tracing::error!(
+                "Failed to deserialize initialization options: {error}. \
+                Using default initialization options."
+            );
+        }
```

Comment:
I'm inclined to move the "fallback to default" logic here or we risk that we someday return something other than the defaults and forget to adjust the log message.